api/api.go

```diff
@@ -423,8 +423,8 @@ func (b *BlockChainAPI) Call(
 
 	res, err := b.evm.Call(ctx, tx, from, cadenceHeight)
 	if err != nil {
-		b.logger.Error().Err(err).Msg("failed to execute call")
-		return handleError[hexutil.Bytes](b.logger, errs.ErrInternal)
+		b.logger.Debug().Err(err).Msg("failed to execute call")
+		return nil, err
```

Comment:
I have removed the use of:

```go
handleError[hexutil.Bytes](b.logger, errs.ErrInternal)
```

as this hides some valuable error messages from the end user.

For example, for the following command:

```bash
curl -XPOST 'localhost:8545' --header 'Content-Type: application/json' --data-raw '{"jsonrpc":"2.0","method":"eth_call","params":[{"from":"0x658bdf435d810c91414ec09147daa6db62406379","to":"0x99466ed2e37b892a2ee3e9cd55a98b68f5735db2","gas":"0x52D4","gasPrice":"0x0","value":"0x0","input":"0xc6888fa10000000000000000000000000000000000000000000000000000000000000006"}, "latest"],"id":4}' | jq
```

Before we would get:

```json
{
  "jsonrpc": "2.0",
  "id": 4,
  "error": {
    "code": -32000,
    "message": "internal error"
  }
}
```

While now, we return:

```json
{
  "jsonrpc": "2.0",
  "id": 4,
  "error": {
    "code": -32000,
    "message": "out of gas"
  }
}
```

This is far more insightful for the user, to figure out what's wrong with the given arguments.

Reply:
I like this idea, but maybe worth checking if it's possible to assert the type of the error is actually an error coming form EVM execution. Now we might return errors that are not relevant for the user.